ambry-protocol/src/main/java/com.github.ambry.protocol/PartitionRequestInfo.java

```diff
@@ -42,7 +42,7 @@ public PartitionRequestInfo(PartitionId partitionId, List<BlobId> blobIds) {
     totalIdSize = 0;
     for (BlobId id : blobIds) {
       totalIdSize += id.sizeInBytes();
-      if (!partitionId.equals(id.getPartition())) {
+      if (!partitionId.toPathString().equals(id.getPartition().toPathString())) {
```

Comment:
Thats why I have this change. In case of a cluster map where local partition comes from static cluster map(Partition.java), while the remote replicas comes from helix cluster map(AmbryPartition.java), it will not to possible to compare them because they will have different types. However, since replication can happen, the blobs returned in replication will have AmbryPartition as partition type. Hence in order to make sure they have the same partition, matching of just the id should suffice.